### PR TITLE
[Function] Implement TAKE function 

### LINKED ISSUE #1273: Implement TAKE function
# Description
Implementation `TAKE()`
Returns the LIST taken only the first n from the LIST

## Example
Function `TAKE()`
```sql
SELECT TAKE(items, size) from Foo;
```

### PR DESCRIPTION
**Pull request by ding-co**
resolves #1273

### CODE REVIEW

File: test-suite/src/function/take.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
+use gluesql_core::executor::EvaluateError;
+
+use {crate::*, gluesql_core::prelude::Value::*};
```
Comment (panarch) [inline test-suite/src/function/take.rs:3]: In our repo, we recommend to wrap imports by using single `use`, then the cargo fmt can automatically do formatting.

File: core/src/executor/evaluate/function.rs [hunk 2]
```diff
@@ -501,6 +501,25 @@ pub fn prepend<'a>(expr: Evaluated<'_>, value: Evaluated<'_>) -> Result<Evaluate
...
+        }
+    };
+
+    match (expr, size) {
+        (Value::List(mut l), _) => {
+            l = l.into_iter().take(size).collect();
+            Ok(Evaluated::Value(Value::List(l)))
+        }
+        _ => Err(EvaluateError::ListTypeRequired.into()),
+    }
```
Comment (panarch) [inline core/src/executor/evaluate/function.rs:520]: 1. `size` does not need to be used in `match` statement.
2. rather than using `mut l`, it would be better to use variable shadowing
then,
```rust
match expr {
    Value::List(l) => {
        let l = l.into_iter().take(size).collect();

        Ok(Evaluated::Value(Value::List(l)))
    }
    _ => Err(EvaluateError::ListTypeRequired.into()),
}
````

File: core/src/plan/expr/function.rs [hunk 3]
```diff
@@ -140,6 +140,7 @@ impl Function {
             }
             | Self::Append { expr, value: expr2 }
             | Self::Prepend { expr, value: expr2 }
+            | Self::Take { expr, size: expr2 }
```
Comment (panarch) [inline core/src/plan/expr/function.rs:143]: ref. https://github.com/gluesql/gluesql/pull/1282/commits/6b4aa8a7cd2d6196110ab0317610263ef66955db

could you add an unit test for this `Self::Take`?

File: core/src/executor/evaluate/function.rs [hunk 4]
```diff
@@ -501,6 +501,25 @@ pub fn prepend<'a>(expr: Evaluated<'_>, value: Evaluated<'_>) -> Result<Evaluate
     }
 }
 
+pub fn take<'a>(expr: Evaluated<'_>, size: Evaluated<'_>) -> Result<Evaluated<'a>> {
```
Comment (panarch) [inline core/src/executor/evaluate/function.rs:504]: you can consider to get `name` param like other functions.
Then the `name` param can be used in `EvaluateError::FunctionRequiresUSizeValue(name)` like errors.

File: test-suite/src/function/take.rs [hunk 5]
```diff
@@ -0,0 +1,61 @@
+use {crate::*, gluesql_core::executor::EvaluateError, gluesql_core::prelude::Value::*};
+
```
Comment (panarch) [inline test-suite/src/function/take.rs:2]: You can consider a few more tests which handles `NULL`
either `expr` or `size` field is null, then the result is null in sql evaluation.
e.g.
```sql
TAKE(NULL, 0)
-- NULL

TAKE(items, NULL)
-- NULL
```

**Review by panarch — CHANGES_REQUESTED** [review]

File: test-suite/src/function/take.rs (hunk 5 above)
Comment (ding-co) [reply test-suite/src/function/take.rs:2]: Null has more high priority than size or list validation?

e.g.
```rust
TAKE(NULL, -5)
-- NULL? or FunctionRequiresUSizeValue?
```

Comment (panarch) [reply test-suite/src/function/take.rs:2]: that's really a good point.
for now, we don't have strict policy about the priority on what to handle earlier, yet.
it may depend on the arrange of implementation codes so it's all up to you.

Comment (ding-co) [reply test-suite/src/function/take.rs:2]: I think about that size or list validation has more priority,
I have to choose one more option to give higher priority validation between them.

So I adapt the way for just returning NULL when NULL value exists in query.

**Review by panarch — APPROVED** [review]
Looks all nice! Thanks a lot for the contribution 👍 👍
